docs/content/ingestion/batch-ingestion.md

```diff
@@ -203,20 +204,25 @@ The configuration options are:
 ```
 
 Single-dimension partitioning works by first selecting a dimension to partition on, and then separating that dimension
-into contiguous ranges. Each segment will contain all rows with values of that dimension in that range. For example,
+into contiguous ranges or different hash value.
+
+If use targetPartitionSize, each segment will contain all rows with values of that dimension in that range. For example,
 your segments may be partitioned on the dimension "host" using the ranges "a.example.com" to "f.example.com" and
 "f.example.com" to "z.example.com". By default, the dimension to use is determined automatically, although you can
 override it with a specific dimension.
 
+If use numShards, all rows with a particular value for that dimension will to go into the same segment.
+
 The configuration options are:
 
 |Field|Description|Required|
 |--------|-----------|---------|
 |type|Type of partitionSpec to be used.|"dimension"|
-|targetPartitionSize|Target number of rows to include in a partition, should be a number that targets segments of 500MB\~1GB.|yes|
+|targetPartitionSize|Target number of rows to include in a partition, should be a number that targets segments of 500MB\~1GB.|either this or numShards|
 |maxPartitionSize|Maximum number of rows to include in a partition. Defaults to 50% larger than the targetPartitionSize.|no|
 |partitionDimension|The dimension to partition on. Leave blank to select a dimension automatically.|no|
 |assumeGrouped|Assume that input data has already been grouped on time and dimensions. Ingestion will run faster, but may choose sub-optimal partitions if this assumption is violated.|no|
+|numShards|specify the number of partitions directly, instead of a target partition size. Ingestion will run faster, since it can skip the step necessary to select a number of partitions automatically.|either this or targetPartitionSize|
```

Comment:
so, now, what is the difference between following configurations?
```
{
  "type": "hashed",
  "numShards": n,
  "partitionDimensions": <whatever>
}
```
and
```
{
  "type": "dimension",
  "numShards": n,
  "partitionDimensions": <whatever>
}
```

I think they will produce exactly same results. In other words, I don't think "dimension" partition spec needs to support "numShards" as it is only used in the case when you want to partition data by dimension value range. Fixed shards case is already supported by "hashed" partition spec.